manifests/service/archive.pp

```diff
@@ -17,29 +17,25 @@
   if $::tomcat::params::systemd {
     # manage systemd unit on compatible systems
     if $::osfamily == 'Suse' { # SuSE
-      file { "${::tomcat::service_name_real} service unit":
-        ensure  => present,
-        path    => "/usr/lib/systemd/system/${::tomcat::service_name_real}.service",
-        owner   => 'root',
-        group   => 'root',
-        content => template("${module_name}/instance/systemd_unit_suse.erb")
-      }
-    } elsif $::operatingsystem == 'Fedora' and $::operatingsystemmajrelease >= '20' { # Fedora 20+
-      file { "${::tomcat::service_name_real} service unit":
-        ensure  => present,
-        path    => "/usr/lib/systemd/system/${::tomcat::service_name_real}.service",
-        owner   => 'root',
-        group   => 'root',
-        content => template("${module_name}/instance/systemd_unit_fedora.erb")
-      }
-    } else { # RHEL 7+ or Fedora 17-19
-      file { "${::tomcat::service_name_real} service unit":
-        ensure  => present,
-        path    => "/usr/lib/systemd/system/${::tomcat::service_name_real}.service",
-        owner   => 'root',
-        group   => 'root',
-        content => template("${module_name}/instance/systemd_unit_rhel.erb")
-      }
+      $systemd_template = "${module_name}/instance/systemd_unit_suse.erb"
+    } elsif $::operatingsystem == 'Fedora' and $::operatingsystemmajrelease < '20' { # Fedora 17-19
+      $systemd_template = "${module_name}/instance/systemd_unit_fedora.erb"
+    } else { # Fedora 20+ or RHEL 7+
+      $systemd_template = "${module_name}/instance/systemd_unit_rhel.erb"
+    }
+    # write service file
+    file { "${service_name_real} service unit":
```

Comment:
I think this is correct. Line 14:
$service_name_real = $::tomcat::service_name_real